sqlmesh/core/context.py

```diff
@@ -1269,8 +1269,11 @@ def plan(
             skip_linter=skip_linter,
         )
 
-        if no_auto_categorization:
+        plan = plan_builder.build()
+
+        if no_auto_categorization or plan.uncategorized:
```

Comment:
FYI: I intentionally did not remove the following check in `apply` as a defensive mechanism: https://github.com/TobikoData/sqlmesh/blob/ec92d47868f03f1b3d29311d7bddbb89c68f89c5/sqlmesh/core/context.py#L1545-L1546